### DISCUSSION TITLE
Downloading and getting a File reference to image

### DISCUSSION THREAD

**Discussion by c-b-h** [coil-kt/coil#1090] [Q&A]
I would like to call something in the lines of this, but with Coil, which we are currently using in our project.

```kotlin
val cacheFile = Glide.with(context)
    .asFile()
    .load(remoteUri)
    .submit()
    .get()
```

Looking at the documentation of Coil, I simply cannot find a way of getting a `File` reference to the loaded image.

**Comment by colinrtwhite** [comment]
In the 2.x alphas you can access it like so `imageLoader.diskCache?.get(url)?.use { /* Read the file using it.data. */ }`

**Reply by c-b-h** [reply]
Thanks for the reply. Is there a way of asking Coil to simply download a file. Your solution seems to use the disk cache which assumes that the image _has_ been fetched already before its `File` being available.

**Reply by wuzebiao** [reply]
I have the same problem

**Reply by yoobi** [reply]
Hello ! Is there a way to do it in the new 2.x ?
EDIT: I found this https://github.com/coil-kt/coil/issues/1353 which is the same answer as above.
